coderd/externalauth/externalauth.go

```diff
@@ -307,8 +300,17 @@ func (c *DeviceAuth) AuthorizeDevice(ctx context.Context) (*codersdk.ExternalAut
 	if err != nil {
 		return nil, err
 	}
+
+	do := http.DefaultClient.Do
+	if c.Cfg != nil {
+		// The cfg can be nil in unit tests.
+		do = func(req *http.Request) (*http.Response, error) {
+			return c.Cfg.Do(ctx, "AuthorizeDevice", req)
+		}
+	}
+
+	resp, err := do(req)
```

Comment:
Noticed we never check HTTP status code for `resp`, should we?

Reply:
I am not sure, I did not write this part. Have not checked the payloads myself here.